app/src/main/java/unipd/se18/ocrcamera/GalleryActivity.java

```diff
@@ -0,0 +1,385 @@
+        //If there is more than 1 fragment running I close the
+        //current one (DetailsFragment) and return to the prevous (HomeFragment)
+        //Or I just call the default onBackPressed which closes the activity
+        if (backStackEntryCount != 0) {
+            fm.popBackStack();
+        } else {
+            super.onBackPressed();
+        }
+    }
+
+
+
+    /************************************************************/
+    /***************************FRAGMENTS************************/
+    /************************************************************/
+
+
+    /**
+     * Fragment of the gallery layout
+     * @author Romanello Stefano
+     */
+    public static class MainFragment extends Fragment
+    {
+        GalleryManager.RecycleCardsAdapter cardAdapter;
+        /**
+         * Event triggered once everything in the activity have finished loading
+         */
+        @Override
+        public void onActivityCreated(Bundle savedInstanceState)
+        {
+            super.onActivityCreated(savedInstanceState);
+            ActionBar actionBar =((GalleryActivity)getActivity()).getSupportActionBar();
+            actionBar.setTitle(R.string.galleryFragmentTitle);
+            actionBar.setDisplayHomeAsUpEnabled(false);
+            actionBar.setHomeButtonEnabled(false);
+        }
+
+        /**
+         * Event triggered when I create the view
+         */
+        @Override
+        public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState)
+        {
+            // Defines the xml file for the fragment
+            return inflater.inflate(R.layout.activity_gallery_content, parent, false);
+        }
+
+        /**
+         * This event is triggered soon after onCreateView().
+         * Any view setup should occur here.
+         */
+        @Override
+        public void onViewCreated(View view, Bundle savedInstanceState)
+        {
+            //Create the container for all my cards
+            RecyclerView picturesRecycleView = view.findViewById(R.id.recycle_view);
+            picturesRecycleView.setHasFixedSize(true);
+
+            //Set a grid (2 columns) from my recycleView
+            GridLayoutManager recycleLayoutManager = new GridLayoutManager(view.getContext(), 2, GridLayoutManager.VERTICAL, false);
+            picturesRecycleView.setLayoutManager(recycleLayoutManager);
+            picturesRecycleView.setItemAnimator(new DefaultItemAnimator());
+
+            //Load the cards using the RecycleCardsAdapter into the picturesRecycleView
+            ArrayList<GalleryManager.PhotoStructure> photos = GalleryManager.getImages(view.getContext());
+            cardAdapter = new GalleryManager.RecycleCardsAdapter(view.getContext(), photos);
+            picturesRecycleView.setAdapter(cardAdapter);
+        }
+
+
+        /**
+         * Event when the main fragment change its hidden status.
+         * @param hidden true=is hidden, false=is not hidden.
+         */
```

Comment:
With "hidden status" the description is general, and a coder rarely see the implementation of the code in order to understand the things that a method does.
Could a good idea be describing the method with a non-general description?

Reply:
I think that "hidden status" means only if an object is hidden or not. I will explain more clearly